wagtail/migrations/0088_fix_log_entry_json_timestamps.py

```diff
@@ -0,0 +1,137 @@
+# Generated by Django 3.2.6 on 2022-11-09 07:50
+
+import datetime
+
+import django.core.serializers.json
+from django.conf import settings
+from django.db import migrations, models
+from django.utils import timezone, dateparse
+
+
+def legacy_to_iso_format(date_string, tz=None):
+    dt = datetime.datetime.strptime(date_string, "%d %b %Y %H:%M")
+    if settings.USE_TZ:
+        dt = timezone.make_aware(dt, datetime.timezone.utc if tz is None else tz)
+        dt = timezone.localtime(dt, datetime.timezone.utc)
+    # We return the datetime object, so DjangoJSONEncoder will serialize it accordingly.
+    return dt
+
+
+def iso_to_legacy_format(date_string, tz=None):
+    dt = dateparse.parse_datetime(date_string)
+    if dt is None:
+        raise ValueError("date isn't well formatted")
+    if settings.USE_TZ:
+        dt = timezone.localtime(dt, datetime.timezone.utc if tz is None else tz)
+    return dt.strftime("%d %b %Y %H:%M")
+
+
+def migrate_logs_with_created_only(model, converter):
+    for item in (
+        model.objects.filter(
+            action__in=["wagtail.revert", "wagtail.rename", "wagtail.publish"]
+        )
+        .only("data")
+        .iterator()
+    ):
+        try:
+            # If a previous_revision was available, the data contains "revision" with
+            # its created date.
+            # Also, there are "wagtail.publish" logs, which don't set data at all.
+            created = item.data["revision"]["created"]
+            # "created" is set to the previous revision's created_at, which is set
+            # to UTC by django.
+            item.data["revision"]["created"] = converter(created)
+        except ValueError:
+            # TODO TBD: log error?
+            continue
+        except KeyError:
+            continue
```

Comment:
I tried running this migration on torchbox.com's Wagtail instance and it failed with a different error: `TypeError: 'NoneType' object is not subscriptable`

The likely cause was that there are old logs where `log.data` is stored as a JSON `null` (not SQL `NULL` or an empty JSON object `{}`) in the database, and it's deserialized into `None` in Python, so accessing `data["revision"]` will fail. More on that in https://github.com/wagtail/wagtail/pull/8021#issuecomment-1658082683.

I think we should guard against `TypeError` here to avoid this. It would also avoid a potentially unlikely issue where `data["revision"]` is `None`.